### ISSUE TITLE
ModuleNotFoundError: No module named 'typing_extensions'

### ISSUE THREAD

**Issue by redjax** [karpetrosyan/hishel#160]
I am trying to follow the documentation, specifically [this section](https://hishel.com/#httpx-and-http-core). I have copied the code like:

```
import httpx
import hishel

transport = httpx.HTTPTransport()
cache_transport = hishel.CacheTransport(transport=transport)

req = httpx.Request("GET", "https://hishel.com")

res1 = cache_transport.handle_request(req)
print(f"[{res1.status_code}: {res1.reason_phrase}]: {res1.text}")
res2 = cache_transport.handle_request(req)
print(f"CACHED [{res2.status_code}: {res2.reason_phrase}]: {res2.text}")

```

When I run this code, I get a `ModuleNotFoundError`:

```
Traceback (most recent call last):
  File "C:\git\Python\Learning\hishel-cache\httpx-hishel\src\httpx_hishel\main.py", line 2, in <module>
    import hishel
  File "C:\git\Python\Learning\hishel-cache\httpx-hishel\.venv\Lib\site-packages\hishel\__init__.py", line 3, in <module>
    from ._async import *
  File "C:\git\Python\Learning\hishel-cache\httpx-hishel\.venv\Lib\site-packages\hishel\_async\__init__.py", line 1, in <module>
    from ._client import *  # noqa: F403
    ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\git\Python\Learning\hishel-cache\httpx-hishel\.venv\Lib\site-packages\hishel\_async\_client.py", line 5, in <module>
    from hishel._async._transports import AsyncCacheTransport
  File "C:\git\Python\Learning\hishel-cache\httpx-hishel\.venv\Lib\site-packages\hishel\_async\_transports.py", line 15, in <module>
    from ._storages import AsyncBaseStorage, AsyncFileStorage
  File "C:\git\Python\Learning\hishel-cache\httpx-hishel\.venv\Lib\site-packages\hishel\_async\_storages.py", line 14, in <module>
    from typing_extensions import TypeAlias
ModuleNotFoundError: No module named 'typing_extensions'
```

I'm not sure if it's the version of `hishel` I've installed, maybe? I only installed `httpx` and `hishel` for this test.

**Comment by redjax** [comment]
Quick comment to add `pip install typing_extensions` fixed the issue. I'm unsure if this is a problem with the `hishel` library, or if it was my environment. I tried in Windows and WSL, and things only worked after adding `typing_extensions`.

Other than this minor issue, this is a great replacement for my custom solution using `DiskCache`! Much easier, thank you for the work :)

**Comment by karpetrosyan** [comment]
Hi @redjax!

> Other than this minor issue, this is a great replacement for my custom solution using DiskCache! Much easier, thank you for the work :)

Thank you :relaxed:

> Quick comment to add pip install typing_extensions fixed the issue. I'm unsure if this is a problem with the hishel library, or if it was my environment. I tried in Windows and WSL, and things only worked after adding typing_extensions.

That was a good catch! We use `typing_extensions`, but it is not required to be installed before `hishel` installation.

We can't easily reproduce this problem because hishel requires httpx, and httpx requires anyio, which would install `typing_extensions` for us. However, in some cases, such as when the Python version is 3.12, anyio would not install `typing_extensions`, resulting in this exception.

So we need to move `typing_extensions` from `requirements.txt` to `pyproject.toml`, making it a normal dependency rather than a development-only dependency.

Would you like to open a pull request resolving this issue?

**Comment by redjax** [comment]
That makes sense. I was running into this with Python 3.11.6, for what it's worth.

I would be happy to do a pull request for this...it will be my first one ever, do you have any instructions? 😬

**Comment by redjax** [comment]
I think I got it: https://github.com/karpetrosyan/hishel/pull/161

**Comment by karpetrosyan** [comment]
Resolved in #161
